test_pytest_html.py

```diff
@@ -25,22 +25,45 @@ def run(testdir, path='report.html', *args):
     return result, html
 
 
-def assert_summary(html, tests=1, duration=None, passed=1, skipped=0, failed=0,
+def assert_results_by_outcome(html, test_outcome, test_outcome_number,
+                              label=None):
+    # Asserts if the test number of this outcome in the summary is correct
+    regex_summary = '(\d)+ {0}'.format(label or test_outcome)
+    int(re.search(regex_summary, html).group(1)) == test_outcome_number
+
+    # Asserts if the generated checkbox of this outcome is correct
+    regex_checkbox = ('<input checked="true" data-test-result="{0}"'
+                      .format(test_outcome))
+    if test_outcome_number == 0:
+        regex_checkbox += ' disabled="true"'
+    assert str(re.search(regex_checkbox, html).group()) != None
```

Comment:
No need to return a string here, and as the flake8 checks point out, comparison to `None` should be `is not None`.